### ISSUE TITLE
Stored XSS within article creation

### ISSUE THREAD

**Issue by prodigysml** [pluxml/PluXml#253]
A writer or the administrator can write javascript within the page, which can then be used to steal other user's session tokens (actually quite possible as the session token doesn't have HttpOnly enabled). Adding something like <img src=x onerror=alert(1)>, in the article's content stores the payload on the server. 
Then any user who visits the page is affected by the payload. 

I would advise HTML encoding the user input and also setting the HttpOnly flag on the session token.

**Comment by bazooka07** [comment]
Yes.
By default, **session.cookie_httponly** is empty in php.ini.

useful link:
http://php.net/manual/en/session.configuration.php#ini.session.cookie-httponly

**Comment by fgeek** [comment]
I don't think it is a issue when administrator can input any JavaScript to the articles, but it's an issue that regular writer can do so or are all writers automatically fully trusted? Missing HttpOnly is separate issue, but should be fixed as well. This also affects Debian unstable.

**Comment by pluxml** [comment]
@bazooka07 as-tu une suggestion pour corriger session.cookie_httponly ?

**Comment by bazooka07** [comment]
Je n'ai pas fait d'essais mais a priori la fonction session_set_cookie_params devrait faire le job avec httponly=true.
http://php.net/manual/fr/function.session-set-cookie-params.php
En profiter pour régler lifetime qui par défaut est égal à 0. 1 à 2H pour rédiger un article devrait suffire.

**Comment by bazooka07** [comment]
En complement du précèdent post, voici un article à lire:
https://phackt.com/xss-cors-csrf-partie-2-xss-cookies-session
Et donc, tester ceci :
```
<?php
const LIFETIME = 2 x 60 x 60; # secondes
session_set_cookie_params(LIFETIME, $plxAdmin->racine.'core/admin/', $_SERVER['SERVER_NAME'], isset($_SERVER["HTTPS"]), true);
?>
```
Autre lien intéressant:
https://php.earth/docs/security/intro
Voir en particulier "Directory traversal (path injection)". A rapprocher de la valeur de PLX_ROOT.
PLX_ROOT est à bannir de toutes les urls.

**Comment by NicoleG25** [comment]
Regarde que [CVE-2017-1001001](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-1001001) assigné à cette vulnérabilité,

était-ce réparé?
@bazooka07

**Comment by P3ter** [comment]
Hi @NicoleG25, I think we can have a first fix (PR #357) for the next PluXml release wich is planned for this week.
